docs/source/developers/guide/introduction.rst

```diff
@@ -0,0 +1,135 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. _introduction:
+
+**********************
+New Contributor's Guide
+**********************
+
+This guide is meant to be a resource for contributing to
+`Arrow <https://arrow.apache.org//>`_ for new contributors.
+
+Starting to contribute to a project like Apache Arrow can look
+scary if not completely impossible. But we can reassure you it is
+not. What we have to do is to take small steps.
+
+
+Why contribute to Arrow?
+========================
+There can be various reasons why someone would want to contribute
+to Arrow and all of them are valid!
+
+* You find the project interesting and would like to give it a go,
+  learn and grow.
+
+* You use Arrow in the project you are working on and it would be
+  great to make some quick fixes by yourself
+
+Read more about the project in the :ref:`about_arrow` section.
+
+Quick Reference
+===============
+
+Here are the basic steps needed to get set up and contribute to Arrow.
+This is meant as a checklist and also to have an overall picture.
+
+For complete instructions please follow :ref:`step_by_step` - a
+step-by-step guide or R and Python :ref:`tutorial-index` for an example
+of adding a basic feature.
+
+
+#. **Install and set up Git, fork the Arrow repository**
+
+   See detailed instructions on how to :ref:`set-up` Git and fork the
+   Arrow repository.
+
+#. **Building Arrow**
+
+   This can be quite easy or a real blocker. Be prepared to dig through the
+   additional flags for cmake in the C++ build guide and read the doc carefully
+   to not miss any suggestions already given there. Anytime you are stuck,
+   reach out to us via appropriate :ref:`communication` channel.
+
+   See detailed instructions on how to :ref:`build-arrow`.
+ 
+#. **Run the tests**
+   
+   Run the tests from a terminal for Python
```

Comment:
Did you mean to use Sphinx tabs here (I see you added the dependency and this would be a reasonable place to use them, especially as other languages may be added).

Reply:
I did, but as there is only one line per language currently, it took much more space as it does now. I think it is better to use tabs when there are more lines of code and the visibility gets improved with using tabs.

Reply:
> I did, but as there is only one line per language currently, it took much more space as it does now. I think it is better to use tabs when there are more lines of code and the visibility gets improved with using tabs.

Did you mean that it takes more space in the rst code here? I think that's fine (I agree sphinx-tabs can be quite verbose), as the majority will look at the built docs online, and there it will not take much more space by having the tabs (certainly if we would also want to add other languages here as well).

Reply:
I meant in the built docs online. The tab is very nice but quite big, visually. It would make sense when we add more languages, I agree.